src/lib/store.js

```diff
@@ -139,6 +174,42 @@ export const reducers = {
       displayLegalModal: false
     }
   ),
+  images: handleActions(
+    {
+      ADD_IMAGE: (
+        state,
+        { payload: { name, importing = true, size, image, type } }
+      ) => ({
+        ...state,
+        images: {
+          ...state.images,
+          [name]: { name, importing, size, image, type }
+        }
+      }),
+      DELETE_IMAGES: (state, { payload: names = [] }) => {
+        const images = { ...state.images };
+        names.forEach(name => delete images[name]);
+        return { ...state, images };
+      },
+      LOAD_IMAGES: (state, { payload: images = {} }) => ({ ...state, images }),
+      // LOAD_IMAGE differs from UPDATE_IMAGE in that the middleware does not
```

Comment:
Kind of a hack to reuse some common code - LOAD_IMAGE is for a local load of an image on localStorage changes between tabs, UPDATE_IMAGE is a more general action that also triggers updates in the add-on